app/view/button/ShowMeasureToolsPanelController.js

```diff
@@ -0,0 +1,145 @@
+Ext.define('MoMo.client.view.button.ShowMeasureToolsPanelController', {
+    extend: 'Ext.app.ViewController',
+
+    requires: [
+    ],
+
+    alias: 'controller.button.showmeasuretoolspanel',
+
+    /**
+     * Placeholder for the measurement tools panel
+     */
+    btnPanel: null,
+
+    /**
+    *
+    */
+    onToggle: function(btn, pressed){
+        var me = this;
+        if (pressed){
+            me.showMeasureToolsPanel();
+        } else {
+            me.hideMeasureToolsPanel();
+            me.deactivateMeasureTools();
+        }
+    },
+
+    /**
+     * Creates a panel containing two buttons for distance and area measurements
+     * The position of the panel will be computed dynamically (s. method
+     * #computePosition below).
+     */
+    createMeasurementButtonsPanel: function() {
+
+        var me = this;
+
+        var parentBtn = me.getView().getEl();
+
+        var position = me.computePosition(parentBtn);
+
+        var btnPanel = Ext.create('Ext.panel.Panel', {
```

Comment:
You're absolutely right. I wanted to refactor the logic after I've added some more tools to the client (to have a better test case). For the moment I leave it as is.